app/src/main/java/com/github/se/polyfit/viewmodel/qrCode/barcodeViewModel.kt

```diff
@@ -47,7 +59,18 @@ class BarCodeCodeViewModel @Inject constructor() : ViewModel() {
     val list = _listId.value?.toMutableList() ?: mutableListOf()
     list.add(0, id)
     _listId.value = list
+    lastScanTime = System.currentTimeMillis()
+    _isScanned.value = true
     Log.v("QrCodeViewModel", "new list: ${_listId.value}")
+
+    CoroutineScope(Dispatchers.IO).launch {
+      delay(TIME_COLOR_BORDER_MILLIS)
+      withContext(Dispatchers.Main) {
+        if (lastScanTime + TIME_COLOR_BORDER_MILLIS < System.currentTimeMillis()) {
+          _isScanned.value = false
+        }
+      }
+    }
```

Comment:
great idea to add a time limit on the bordure color change but I would suggest using `_isScanned.postValue(false)` as it is a thread safe function to modify the value of `LiveData`